src/get-env-variables.js

```diff
@@ -0,0 +1,60 @@
+const exec = require('child_process').exec
+// import spawn from 'spawn-command-with-kill'
+import {assign, clone} from 'lodash'
+import managePath from 'manage-path'
+import {sync as findUpSync} from 'find-up'
+
+const NON_ERROR = 0
+
+export default getEnv
+
+function getEnv() {
+  return new Promise((resolve, reject) => {
```

Comment:
You can rewrite this by simply chaining the promise. You don't need to use the promise constructor, you can simply do:

```javascript
return getNpmEnvVariables().then(npmVars => {
  const env = assign(clone(process.env), npmVars)
  const alterPath = managePath(env)
  const npmBin = findUpSync('node_modules/.bin')
  if (npmBin) {
    alterPath.unshift(npmBin)
  }
  return env
})
```